src/ir/item.rs

```diff
@@ -1444,6 +1445,15 @@ impl ItemCanonicalPath for Item {
                                       ctx: &BindgenContext)
                                       -> Vec<String> {
         let path = self.canonical_path(ctx);
+        if let super::item_kind::ItemKind::Type(ref type_) = self.kind {
+            if let &TypeKind::Enum(ref enum_) = type_.kind() {
+                if enum_.is_constified_enum_module(ctx, self) {
+                    // Type alias is inside a module
+                    return vec![path.last().unwrap().clone(),
```

Comment:
This probably won't work for C++ namespaces, you need to use `let mut path` above and `path.push(CONSTIFIED_ENUM_MODULE_REPR_NAME.into())` here, then keep executing the rest of the logic of this function.

Though it feels somewhat hacky to put it here...